snapcraft/tests/test_meta.py

```diff
@@ -317,6 +334,41 @@ def test_create_meta_with_vararg_config(self, mock_the_open,
             ),
         ])
 
+    @patch('snapcraft.meta._write_wrap_exe')
+    @patch('snapcraft.meta.open', create=True)
+    def test_create_meta_with_license(self, mock_the_open, mock_wrap_exe):
+        self.config_data.pop('config')
+        self.config_data['license'] = 'LICENSE'
+
+        meta.create(self.config_data)
+
+        self.mock_makedirs.assert_has_calls([
+            call(self.meta_dir, exist_ok=True),
+            call(self.hooks_dir),
+        ])
+
+        self.mock_rename.assert_not_called()
+
+        mock_the_open.assert_has_calls(self.expected_open_calls)
+        mock_wrap_exe.assert_has_calls([
+            call(
+                '$SNAP_APP_PATH/bin/bash',
+                os.path.join(os.path.abspath(os.curdir),
+                             'snap/bin/bash.wrapper'),
+                args=None, shebang=None,
+            ),
+        ])
+
+        self.mock_copyfile.assert_has_calls([
+            call('my-icon.png', os.path.join(self.meta_dir,
+                 'my-icon.png')),
+            call('file.apparmor', os.path.join(self.meta_dir,
+                 'file.apparmor')),
+            call('file.seccomp', os.path.join(self.meta_dir,
+                 'file.seccomp')),
```

Comment:
These three calls are not related to the license. You can use mock.ANY instead, I think here we only care about the license call.